DependencyInjection/CmfSeoExtension.php

```diff
@@ -267,15 +267,18 @@ private function loadSitemapHandling($config, XmlFileLoader $loader, ContainerBu
                     'priority' => 1,
                 ));
                 $container->setDefinition($this->getAlias().'.sitemap.guesser.'.$key.'.default_change_frequency', $definition);
-                unset($configurations[$key]['default_change_frequency']);
             }
+            unset($configurations[$key]['default_change_frequency']);
+
             if (isset($config['defaults']['templates'])) {
                 // copy default configuration into this sitemap configuration to keep controller simple
                 foreach ($config['defaults']['templates'] as $format => $name) {
-                    if (!isset($configurations['templates'][$format])) {
-                        $configurations['templates'][$format] = $name;
+                    if (!isset($configurations[$key]['templates'][$format])) {
+                        $configurations[$key]['templates'][$format] = $name;
```

Comment:
the configuration test seems to be incomplete, as this mistake was not detected.